InpaintHTTP/MainMod.cs

```diff
@@ -1,111 +1,174 @@
+
+                        if (BitmapImg == null)
+                        {
+                            BitmapImg = tempBitmap;
+                        }
+                        else if (BitmapMask == null)
+                        {
+                            BitmapMask = tempBitmap;
+                        }
+                        else
+                        {
+                            donors.Add(tempBitmap.ToArgbImage());
+                        }
                     }
-                    else {
-                        donors.Add(tempBitmap.ToArgbImage());
+
+                    var imageArgb = ConvertToArgbImage(BitmapImg);
+                    var markupArgb = ConvertToArgbImage(BitmapMask);
+
+                    var inpainter = new Inpainter();
+
+                    // NOTE: This should no longer be used since the creation of InpaintSettings comes with default variables itself?
+                    //var settings = new InpaintSettings
+                    //{
+                    //    MaxInpaintIterations = 15,
+                    //    PatchDistanceCalculator = ImagePatchDistance.Cie76,
+                    //    PatchSize = 11
+                    //};
+
+                    // Convert body request to settings
+                    InpaintSettings userSettings = new InpaintSettings();
+                    try
+                    {
+                        userSettings = JsonConvert.DeserializeObject<InpaintSettings>(Request.Body.AsString());
                     }
-                }
+                    catch { }
 
-                var imageArgb = ConvertToArgbImage(BitmapImg);
-                var markupArgb = ConvertToArgbImage(BitmapMask);
+                    // NOTE: Now merge then? giving priority to the default one?
 
-                var inpainter = new Inpainter();
-                var settings = new InpaintSettings
-                {
-                    MaxInpaintIterations = 15,
-                    PatchDistanceCalculator = ImagePatchDistance.Cie76,
-                    PatchSize = 11
-                };
-                if (!Int32.TryParse(Environment.GetEnvironmentVariable("MAX_INPAINT_ITERATIONS"), out settings.MaxInpaintIterations))
-                {
-                    settings.MaxInpaintIterations = 15;
-                }
-                string patchDistanceEnvVar = Environment.GetEnvironmentVariable("PATCH_DISTANCE_CALCULATOR");
-                if (patchDistanceEnvVar != null && patchDistanceEnvVar.Equals("Cie2000", StringComparison.OrdinalIgnoreCase)) {
-                  settings.PatchDistanceCalculator = ImagePatchDistance.Cie2000;
-                }
-                int patchSize;
-                if (Int32.TryParse(Environment.GetEnvironmentVariable("PATCH_SIZE"), out patchSize))
-                {
-                    settings.PatchSize = (byte)patchSize;
-                }
+                    if (userSettings.PatchSize > ApiSettings._Instance.InpaintSettings.PatchSize)
```

Comment:
Null reference exception here. Sine you forgot to initialize the `AppSettings._Instance` ;)

Reply:
Fixed

Reply:
Added aditional check to prevent `GetEnvironmentVariable` from setting the value to `0`, this seems to give another null ref for the `finalResult`.